java/src/main/native/include/jni_utils.hpp

```diff
@@ -296,15 +296,15 @@ class native_jArray {
     return data_ptr;
   }
 
-  const N_TYPE* const begin() const { return data(); }
+  N_TYPE const* begin() const { return data(); }
 
   N_TYPE* begin() { return data(); }
 
-  const N_TYPE* const end() const { return data() + size(); }
+  N_TYPE const* end() const { return data() + size(); }
 
   N_TYPE* end() { return data() + size(); }
 
-  const J_ARRAY_TYPE get_jArray() const { return orig; }
+  J_ARRAY_TYPE const get_jArray() const { return orig; }
```

Comment:
Below we're removing const from the array type result (e.g.: native_jpointerArray has get_jArray() return a jlongArray, not jlongArray const) but this preserves the const, seems inconsistent.  For example, native_jlongArray instantiates this template with J_ARRAY_TYPE == jlongArray.

Reply:
Not sure why this does not trigger compiler warnings while the removed ones did.